consensus/tool/src/main.rs

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! A cli tool for interrogating a mobilecoin consensus node.
+
+#![deny(missing_docs)]
+
+use clap::{Parser, Subcommand};
+use grpcio::{ChannelBuilder, EnvBuilder};
+use mc_common::logger::{create_app_logger, o};
+use mc_consensus_api::{consensus_common_grpc::BlockchainApiClient, empty::Empty};
+use mc_util_grpc::ConnectionUriGrpcioChannel;
+use mc_util_uri::ConsensusClientUri;
+use std::{sync::Arc, time::Duration};
+
+/// Command line config, set with defaults that will work with
+/// a standard mobilecoind instance
+#[derive(Clone, Debug, Parser)]
+#[clap(
+    name = "mc-consensus-tool",
+    about = "A cli tool for interrogating one or more consensus nodes"
+)]
+pub struct Config {
+    /// Command to use
+    #[clap(subcommand)]
+    pub tool_command: ToolCommand,
+
+    /// Consensus_uri's to connect to
+    #[clap(global = true)]
+    pub consensus_uris: Vec<ConsensusClientUri>,
+}
+
+/// Commands that the tool recognizes
+#[derive(Clone, Debug, Subcommand)]
+pub enum ToolCommand {
+    /// Status: Prints the network status, including block count and block
+    /// version
+    Status,
+    /// Wait-for-quiet: Blocks until the network is 'quiet'. This means that the
+    /// block count doesn't move for some time.
+    WaitForQuiet {
+        /// Number of seconds the network must stop moving for. Defaults to 20.
+        #[clap(long)]
+        period: Option<u64>,
+    },
+}
+
+const DEFAULT_PERIOD_SECONDS: u64 = 20;
+
+fn main() {
+    let (logger, _global_logger_guard) = create_app_logger(o!());
+    let config = Config::parse();
+
+    let env = Arc::new(EnvBuilder::new().name_prefix("consensus-tool-grpc").build());
+
+    let blockchain_conns: Vec<_> = config
+        .consensus_uris
+        .iter()
+        .map(|uri| {
+            let ch =
+                ChannelBuilder::default_channel_builder(env.clone()).connect_to_uri(uri, &logger);
+            BlockchainApiClient::new(ch)
+        })
+        .collect();
+
+    match config.tool_command {
+        ToolCommand::Status => {
+            for conn in &blockchain_conns {
+                let last_block_info = conn
+                    .get_last_block_info(&Empty::new())
+                    .expect("get last block info");
+                println!("{:?}", last_block_info)
+            }
+        }
+        ToolCommand::WaitForQuiet { period } => {
+            let mut last_block_index = 0u64;
```

Comment:
I think there should be an attempt to initialize this to the current block index of one of the nodes. Otherwise, I believe we will often will sleep at least twice the period amount, since was_updated will always change to true at the first iteration (since the last_block_index will very likely be >0)

Reply:
i think that's actually as intended here -- it works better for the tests anyways if we sleep for some period before actually waiting for quiet. In one iteration, I had period set to 5 and we would still see test failures where the entire "wait for quiet" command would happen before a mint config tx had gone by

Reply:
maybe there should be an extra argument that is like `--expect-block-index-greater-than X` or something like this

Reply:
> i think that's actually as intended here -- it works better for the tests anyways if we sleep for some period before actually waiting for quiet. In one iteration, I had period set to 5 and we would still see test failures where the entire "wait for quiet" command would happen before a mint config tx had gone by

This sounds like maybe the period needed to be higher?
I think the expected behavior is if you request a quiet time of 10 seconds for example, and the network is all currently at some non-zero block and is inactive, running the script should take 10 seconds and not 20.

Reply:
alright, the next PR will be like this